### PR TITLE
substr to startsWith conversions

### PR DESCRIPTION
**Pull request by neheb**
Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

**Review by piponazoSchindler — APPROVED** [review]
LGTM. I did not know about the featuring testing in c++ 😱

File: src/utils.hpp [hunk 1]
```diff
@@ -6,8 +6,13 @@
 
 namespace Exiv2::Internal {
 
-constexpr bool startsWith(std::string_view s, std::string_view start) {
+template <typename T, typename Y>
```
Comment (piponazoSchindler) [inline src/utils.hpp:9]: With the change to templates, now some calls to this function replace `T` with `const char *`

**Review by piponazo — APPROVED** [review]

File: src/utils.hpp (hunk 1 above)
Comment (neheb) [reply src/utils.hpp:9]: Hmm? That would be a hard compilation error.

Comment (piponazo) [reply src/utils.hpp:9]: Exactly, like the ones we are getting on CI 😄 
https://github.com/Exiv2/exiv2/actions/runs/4059181741/jobs/6986956283

**Comment by neheb** [comment]
I'll retire this for now.

**Pull request closed without merging**